src/kontrol/prove.py

```diff
@@ -63,14 +63,20 @@ def foundry_prove(
 
     foundry.mk_proofs_dir()
 
-    summary_ids = (
-        [
-            foundry.get_apr_proof(include_summary.id).id
-            for include_summary in collect_tests(foundry, include_summaries, reinit=False)
-        ]
-        if include_summaries
-        else []
-    )
+    summary_ids: list[str] = []
+    if prove_options.cse:
+        test_suite = collect_tests(foundry, tests, reinit=prove_options.reinit, return_empty=True)
+        for test in test_suite:
+            if not isinstance(test.method, Contract.Method) or test.method.function_calls is None:
+                continue
+
+            test_version_tuples = [parse_test_version_tuple(t) for t in test.method.function_calls]
```

Comment:
Here, I would include only functions that are not already in `summary_ids` (functions that have not been proved already).